### ISSUE TITLE
I have "CANT_SUMMON_THE_DAEMON" error 

### ISSUE THREAD

**Issue by leillo1975** [kozec/sc-controller#459]
I can't use my Steam controller. This is the screenshot of this error:

`<blockquote class="imgur-embed-pub" lang="en" data-id="a/JqkAq3Q"><a href="//imgur.com/JqkAq3Q"></a></blockquote><script async src="//s.imgur.com/min/embed.js" charset="utf-8"></script>`

**Comment by kozec** [comment]
> D Mapper Creating virtual devices

If that's last thing it outputs, it's stuck on communicating with uinput (kernel module). Assuming it worked before, check if there is anything uinput related in `dmesg` and either try unloading and reloading it or just reboot.

**Comment by LinuxJoeUser** [comment]
I'd been experiencing this same error code more and more frequently, to the point that I would have to reboot after each use of sc-controller.  

I was thinking it has something to do with scc-daemon being shut down, since I would have no problem running SCC once after a reboot, but couldn't do it again after that.  On a hunch I unchecked the "Automatically Disable Emulation When Closing GUI" option in Settings > Advanced and---since the daemon isn't being shut down---the problem doesn't occur.

I've been trying to find a deeper pattern before making a report here, but since it's now been reported, I figured I'd chime in.  Here's `scc-daemon debug` output:

  ```~$ scc-daemon debug
D SCCDaemon     Starting SCCDaemon...
D SCCDaemon     Initializing drivers...
W SCCDaemon     Skipping disabled driver 'ds4drv'
W SCCDaemon     Skipping disabled driver 'fake'
W SCCDaemon     Skipping disabled driver 'remotepad'
W SCCDaemon     Skipping disabled driver 'yokedrv'
D USB           Registered USB driver for 28de:1102
D USB           Registered USB driver for 28de:1142
D Mapper        Creating virtual devices
Exception AttributeError: "'Keyboard' object has no attribute '_fd'" in <bound method Keyboard.__del__ of <scc.uinput.Keyboard object at 0x7f1430cb2b90>> ignored
```

I looked at `dmesg` output, but I don't know what I'm looking for, honestly.  Xubuntu 19.04 is due out in a few days, and I'm wondering if all this will go away with an upgrade, or---failing that---a clean install.

Let me know if I can help further.

**Comment by kozec** [comment]
Problem is that what you are getting is symptom, not the issue. Issue is that uinput module stops responding. sc-controller is definitely main suspect here, but it may be caused by any other application that uses uinput (including Steam) as well. One of ways how to reproduce the issue is to enable rumble emulation in either of them and then let SystemD to kill them in middle of communication with kernel on logout.

Anyway, if you have rumble emulation turned on, try disabling it for while. It's only case when sc-controller communicates with uinput in way that may get it stuck. If that helps, we know what's happening, if not, I'll try to guess something else :)

**Comment by LinuxJoeUser** [comment]
I don't have rumble emulation enabled, *but* let's not worry about this for now.  I had only mentioned it because I thought it was part of the same issue as leillo1975's.  

I'm going to be doing a clean install when I upgrade to Xubuntu 19.05, and that will happen sometime in the next few days. That might well clean this issue up for me.   If this problem still exists, then I'll let ya know.  Until then keeping scc-daemon running in the background is a fine workaround.

As far as I'm concerned you can close this issue.  :)  Thanks

**Comment by LinuxJoeUser** [comment]
Addendum:  Did a clean install and sure enough this issue came up again.  I figured out that if I disabled *Valve Steam Controller* in Xfce's settings the problem went away.  This turns off (kernel support?) for the SC controller it seems.  I'm thinking this should be a new issue, yeah?

**Comment by kozec** [comment]
Yes, it would be funny if kernel ended up interfering with scc :(

Where is that setting you've turned off located?

**Comment by LinuxJoeUser** [comment]
Interesting twist:  "Valve Steam Controller," which I disabled,  has been replaced by sc-controller, which is enabled.  So maybe it's not a kernel thing?   Anyway, here's the setting location:  Settings Manager > Mouse / Trackpad.  I'm running Xubuntu 19.04

![Peek 2019-04-24 09-57](https://user-images.githubusercontent.com/23580979/56665170-86c96e00-6677-11e9-9de3-a65a4b18671e.gif)

**Comment by DaiKaiser93** [comment]
Just received the same error
Here's scc-daemon debug output:

> sudo scc-daemon debug
D SCCDaemon     Starting SCCDaemon...
D SCCDaemon     Initializing drivers...
W SCCDaemon     Skipping disabled driver 'fake'
W SCCDaemon     Skipping disabled driver 'remotepad'
D USB           Registered USB driver for 054c:09cc
D USB           Registered USB driver for 28de:1102
D USB           Registered USB driver for 28de:1142
D Mapper        Creating virtual devices
D Mapper        Keyboard: <scc.uinput.Keyboard object at 0x7fa476b7a410>
D Mapper        Mouse:    <scc.uinput.Mouse object at 0x7fa4769e3890>
D Mapper        Gamepad:  <scc.uinput.UInput object at 0x7fa4769e39d0>
D SCCDaemon     Created control socket /root/.config/scc/daemon.socket
D SCCDaemon     Connected to XServer :0
Failed to open device: [Errno 16] Device or resource busy
D OSD DaemonCtrl    Connected to daemon, version 0.4.7
D OSD DaemonCtrl    Daemon is ready.
I SCCDaemon     Registered scc-osd-daemon
I OSD osd.daemon    Sucessfully registered as scc-osd-daemon

Can only use my Dualshock 4 via USB, for bluetooth to work I have to run it with sudo sc-controller  via terminal.

**Comment by Galcian79** [comment]
I had the same issue today on Linux Mint 19.3 and i can confirm that turning off rumble support will fix the issue. 
I suspect it is not related to scc but to sc firmware in which rumble emulation is still at an experimental stage. I say this because i noticed similar issues on Steam Windows when enabling it: rumbling getting stuck, game hanging, steam controller turning off. https://steamcommunity.com/app/353370/discussions/0/1732089092442358011/
I suggest to remove the rumble support for now, at least until Valve fixes that (if they ever will do). Or at least not enabling it by default and warning users about potential issues.

**Comment by Galcian79** [comment]
I think i finally managed to reproduce the issue, kind of.
So if i log into the desktop, launch the sc-controller gui, wait for the green lamp, turn on the Steam controller = GOOD.
Logging in, turning on the Steam Controller, red lamp = BAD.
So it seems that, for some reason, the sc daemon works for me only if i launch the gui prior to turning on the Steam controller. Strange behavior indeed.
But i managed a workaround. I followed this post: https://askubuntu.com/questions/663187/how-can-i-run-a-program-on-startup-minimized to create a script for startup manager that launches sc-controller gui and minimizes it without even notice. Works like a charm! :P

**Comment by dveragon** [comment]
I too have this issue but for the controller simple doesn't start the deamon as long as i start it manually it works just fine to note is the i tested it with the 0.4.9.9 version after it happened with 0.4.9.10 (where rhis issue happened for me) and it didn't come up. So i only get this issue with 0.4.9.10 while 0.4.9.9 works just fine
